R/ml_random_forest.R

```diff
@@ -41,11 +53,41 @@ ml_random_forest <- function(x,
     ml.options = ml.options
   )
 
+  col.sample.rate <- ensure_scalar_double(col.sample.rate, allow.null = TRUE)
+  if (!is.null(col.sample.rate)) {
+    if (!(col.sample.rate > 0 && col.sample.rate <= 1))
+      stop("'col.sample.rate' must be in (0, 1]")
+    if (spark_version(sc) < "2.0.0") {
+      if (col.sample.rate == 1)
+        col.sample.rate <- "all"
+      else {
+        # Prior to Spark 2.0.0, random forest does not support arbitrary
+        #   column sampling rates. So we map the input to one of the supported
+        #   strategies: "onethird", "sqrt", or "log2".
+        k <- length(features)
+        strategies <- dplyr::data_frame(strategy = c("onethird", "sqrt", "log2"),
+                                        rate = c(1/3, sqrt(k)/k, log2(k)/k)) %>%
+          arrange(!! rlang::sym("rate"))
```

Comment:
Ahh, good point. One can resolve this using e.g. `utils::globalVariables()` but that's a pretty ugly hammer to wield here. Given that this is to avoid R CMD check issues I think this is fine.